torch/_inductor/codegen/triton.py

```diff
@@ -2574,11 +2589,17 @@ def define_kernel(self, src_code, node_schedule):
             wrapper.src_to_kernel[src_code] = kernel_name
             subs_name = kernel_name if config.triton.unique_kernel_names else "triton_"
 
+            op_info = None
+            if node_schedule and config.triton.descriptive_names:
+                op_info = get_origin_op_info(
+                    node_schedule, config.triton.descriptive_names
+                )
             # DESCRIPTIVE_NAME is used for profiling purposes; it shows the full kernel name
             # even when unique_kernel_names is turned off. Meanwhile, KERNEL_NAME is sometimes set
             # to "triton_" to maximize caching opportunities (when unique_kernel_names = False).
             src_code = src_code.replace(str(Placeholder.DESCRIPTIVE_NAME), kernel_name)
             src_code = src_code.replace(str(Placeholder.KERNEL_NAME), subs_name)
+            src_code = src_code.replace(f"'{Placeholder.ORIGIN_INFO}'", str(op_info))
```

Comment:
Hi @davidberard98 I noticed while getting regressions to pass that these Placeholders are used to to prevent triton from generating multiple copies of the same kernel if the operator instance resolves to the same kernel implementation as a previous operator instance.  See this test [test_dropout3](https://github.com/pytorch/pytorch/blob/main/test/inductor/test_torchinductor.py#L5704-L5710).  It has 2 instances of a dropout op.  The test expects to see 1 dropout kernel which is called once per drop out instance.  When we put information in the meta attribute of the triton kernel, like the kernel name with an instance id, it will cause a new kernel instance to be generated.  I see the placeholders are used to prevent this from happening, but you only get 1 kernel instance recorded, so you're losing information about the 2nd kernel instance.
In my case I want to see information specific to each instance of the kernel.  The origin_op info is showing specifically which operator instances in the network graph were fused together for this specific instance of the kernel.   So I think I need to capture the op info at the kernel call, not the kernel definition.

Reply:
hey, thanks for the info (and sorry for the delayed review!)

> but you only get 1 kernel instance recorded, so you're losing information about the 2nd kernel instance.
is this still the case when descriptive names are turned on?

if you go the route of capturing the info at the kernel call, do you have any ideas how to do that?

Reply:
Oh ok let me try with descriptive names turned on.
So far the only way I know to capture info at the time of kernel call is to insert an nvtx_range push/pop.  
I'll get back to you soon, I'm kind of swamped at work right now so my responses will be delayed.

Reply:
Changing the value of descriptive_names and unique_kernel_names didn't have any effect on the number of instances of a kernel.  I'd like to just go with the current approach and live with inserting the marker at the kernel definition rather than the kernel call.